tensorflow_datasets/core/features/image_feature.py

```diff
@@ -83,17 +86,26 @@ def __init__(self, shape=None, encoding_format=None):
         images on disk.
         If image is loaded from {bmg,gif,jpeg,png} file, this parameter is
         ignored, and file original encoding is used.
+      dtype: tf.uint16 or tf.uint8 (default).
 
     Raises:
       ValueError: If the shape is invalid
     """
     self._encoding_format = None
     self._shape = None
     self._runner = None
+    self._dtype = None
 
     # Set and validate values
     self.set_encoding_format(encoding_format or 'png')
     self.set_shape(shape or (None, None, 3))
+    self.set_dtype(dtype or tf.uint8)
+
+  def set_dtype(self , dtype):
+    """ Update the dtype. """
+    if dtype not in ACCEPTABLE_DTYPES:
```

Comment:
np.uin16 can only be used for png. Please also add a test for this. Also specify this in the `__init__` args section.

Reply:
@Conchylicultor  thanks for the reviewing and mentioning the changes , I made the requested changes , but couldn't add the test you mentioned (i.e. Test case to check np.uint16 is used only with 'png') , please help me with this. 
Thank you!!

Reply:
I don't understand. Why can't you add a new test here:
https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/core/features/image_feature_test.py

```python
self.assertFeature(
        feature=features_lib.Image(..., dtype=tf.uint16),
        shape=(None, None, 3),
        dtype=tf.uint16,
        tests=[
           ...
        ],
)
```